### DISCUSSION TITLE
Do we need to install prisma-client as dependency?

### DISCUSSION THREAD

**Discussion by tarunregmi** [prisma/prisma#24910] [Q&A]
### Question

I'm new to Prisma and here's what I found:

If we install prisma-client as,

- **dependency**, it works in both (development and production) environments because it exists in both environments.
- **devDependency**, it works in development but **not** in production because it is removed by npm.
- **devDependency** and use a **custom folder** for prisma-client, it works in both environments.

Prisma-client as dependency and size in:

- development: 98.9 MB (work)
- production: 71.4 MB (work)

Prisma-client as devDependency and size in:

- development: 98.9 MB (work)
- production: 77.7 kB (**not** work)

Prisma-client as devDependency and custom folder for prisma-client and size in:

- development: 116.2 MB (work)
- production: 17.4 MB (work)

**Note**: Here, size is the total size of the project including the node_modules folder.

Now which approach I should follow?

1. **Default** (use prisma-client as dependency and use default folder for prisma-client)
2. **Another** (use prisma-client as devDependency and use custom folder for prisma-client)
3. **Other** (that I don't know)

Sample project: https://github.com/tarunregmi/prisma-crash

**Comment by ludralph** [comment]
Hi @tarunregmi 👋 

The recommended setup is to have `prisma` as a devDependency and `@prisma/client` as a dependency. This ensures that the Prisma CLI, which is used for development tasks like generating the Prisma Client and running migrations, is only included in your development environment. Meanwhile, the Prisma Client, which is used to interact with your database, is included in both development and production environments.

To set this up, you would run the following commands:
```
npm install @prisma/client
npm install prisma --save-dev
```

- @prisma/client: This is the Prisma Client library that you use in your application code to interact with your database. It should be installed as a regular dependency so that it is available in both development and production environments.

- prisma: This is the Prisma CLI, which you use during development to generate the Prisma Client, run migrations, and other tasks. It should be installed as a devDependency.

**Reply by tarunregmi** [reply]
Thanks @ludralph ,

But my concern is similar to the issue raised here in #17720

**Reply by ludralph** [reply]
Hi @tarunregmi 

I see you already shared a workaround in that issue. Our team will need to investigate this further and get back to you on it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
